script.js

```diff
@@ -1,4 +1,109 @@
+//Win-Lose-Draw Record
+var userCount = 0;
+var opponentCount = 0;
+var drawCount = 0;
+var totalRounds = userCount + opponentCount + drawCount;
+var myWinningPercentage = (userCount / totalRounds) * 100;
+
+var currentGameMode = `Waiting for Username`;
+var userName = ``;
+
+var diceRoll = function () {
+  //Generate a number between 0 to 3
+  var randomDecimal = Math.random() * 3;
+
+  //Remove decimal
+  var randomNumber = Math.floor(randomDecimal);
+
+  console.log("Opponent Rolled");
+  console.log(randomNumber);
+
+  //Assigning an elements to values 0 to 3
+  if (randomNumber == 0) {
+    return "scissors";
+  }
+  if (randomNumber == 1) {
+    return "paper";
+  }
+  if (randomNumber == 2) {
+    return "stone";
+  }
+};
+
+//Define Input
 var main = function (input) {
```

Comment:
great work implementing the game logic. I think a suggestion will be to split up your main function into smaller helper functions to be called in the main function. So perhaps you winning logic could be its own function. The reason for this is so that your code is much more readable and a general rule of thumb for functions is also that they should be less than one full screen